### PR TITLE
MM-28803 - Cloud settings defaults

### PR DESCRIPTION
**Pull request by marianunez**
#### Summary
Update settings defaults specifically for cloud **ONLY on `cloud_beta` branch**. Ticket for the server to support setting defaults based on license [MM-28883](https://mattermost.atlassian.net/browse/MM-28883). 

**Settings changed:**
- Max Users for team: `1000`
- EnableCustomEmoji: `true`
- ExperimentalSideBar: `default_on`
- EnableBotAccountCreation: `true`
- RequireEmailVerification: `true`
- RestrictSystemAdmin: `true`
- TeammateNameDisplay: `FullName`

- ExperimentalGossip:
	- UseExperimentalGossip: `true`
	- EnableExperimentalGossipEncryption: `true`
- SqlSettings:
	- MaxIdleConns:`10`
	- MaxOpenConns `100`
- Email:
	- FeedbackEmail: `feedback@mattermost.com`
	- ReplyToAddress: `noreply@mattermost.com`
	- ConnectionSecurity: `TLS`

#### Ticket Link
[MM-28803](https://mattermost.atlassian.net/browse/MM-28803)

### CODE REVIEW

**Comment by marianunez** [comment]
Some of the defaults here are already being overridden by environment variables in the provisioner.

However, some of the ones overridden are still shown in the System Console with a message that has been overridden and the admin can not change. We should either remove the env var or hide them from the System Console. Those settings are: 

- ExperimentalSideBar: `default_on`
- EnableEmailInvitations: `true`
- RequireEmailVerification: `true`
- SendEmailNotifications: `true` (overridden with the same default value and just disables it in the System Console). 
- ExperimentalViewArchivedChannels: `true` (overridden with the same default value and just disables it in the System Console). 

0/5 if we keep the SQL, email and gossip settings just in the env vars, and remove from this PR.

File: model/config.go [hunk 1]
```diff
@@ -679,7 +674,7 @@ func (s *ServiceSettings) SetDefaults(isUpdate bool) {
 	}
 
 	if s.ExperimentalChannelSidebarOrganization == nil {
-		s.ExperimentalChannelSidebarOrganization = NewString("disabled")
+		s.ExperimentalChannelSidebarOrganization = NewString("default_on")
```
Comment (marianunez) [inline model/config.go:677]: See https://github.com/mattermost/mattermost-server/pull/15540#issuecomment-695106160

**Review by jwilander — APPROVED** [review]
Looks good 👍

**Review by ethervoid — APPROVED** [review]
I'm seeing that we're only changing this in the `cloud-beta` branch so it LGMT

**Comment by jwilander** [comment]
My guess is tests are failing because they expect certain defaults

**Comment by marianunez** [comment]
> My guess is tests are failing because they expect certain defaults

Correct, been resetting the defaults for some tests but more keep popping up. 😩

**Comment by marianunez** [comment]
> Some of the defaults here are already being overridden by environment variables in the provisioner.
> 
> However, some of the ones overridden are still shown in the System Console with a message that has been overridden and the admin can not change. We should either remove the env var or hide them from the System Console. Those settings are:
> 
> * ExperimentalSideBar: `default_on`
> * EnableEmailInvitations: `true`
> * RequireEmailVerification: `true`
> * SendEmailNotifications: `true` (overridden with the same default value and just disables it in the System Console).
> * ExperimentalViewArchivedChannels: `true` (overridden with the same default value and just disables it in the System Console).
> 
> 0/5 if we keep the SQL, email and gossip settings just in the env vars, and remove from this PR.

@joram thoughts on the above? Remove from env vars or hide in System Console?

**Comment by jwilander** [comment]
I'd say hide in system console. I don't think we want cloud admins to ever be able to turn those settings off.

**Pull request merged**